src/main/java/com/shaft/tools/listeners/CucumberFeatureListener.java

```diff
@@ -0,0 +1,113 @@
+package com.shaft.tools.listeners;
+
+import com.shaft.cli.FileActions;
+import com.shaft.gui.element.ElementActions;
+import com.shaft.tools.io.PropertiesFileManager;
+import com.shaft.tools.io.ReportManager;
+import io.cucumber.core.feature.FeatureParser;
+import io.cucumber.core.gherkin.Feature;
+import io.cucumber.core.resource.Resource;
+import io.cucumber.plugin.ConcurrentEventListener;
+import io.cucumber.plugin.event.*;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.function.Supplier;
+
+public class CucumberFeatureListener implements ConcurrentEventListener {
+    @Override
+    public void setEventPublisher(EventPublisher publisher) {
+        //https://github.com/cucumber/cucumber-jvm/issues/1901
+        //https://github.com/cucumber/cucumber-jvm/issues/1901#issuecomment-600494342
+        publisher.registerHandlerFor(TestRunStarted.class, this::handleTestRunStarted);
+        publisher.registerHandlerFor(TestRunFinished.class, this::handleTestRunFinished);
+        publisher.registerHandlerFor(TestCaseStarted.class, this::handleTestCaseStarted);
+        publisher.registerHandlerFor(TestStepStarted.class, this::handleTestStepStarted);
+        publisher.registerHandlerFor(TestSourceParsed.class, this::handleTestSourceParsed);
+    }
+
+    private void handleTestRunStarted(TestRunStarted event) {
```

Comment:
Codacy found an issue: [Avoid unused method parameters such as 'event'.](https://app.codacy.com/manual/mohab.mohieeldeen/SHAFT_ENGINE/pullRequest?prid=5731302)

Reply:
it's mandatory